front/components/product/product.html

```diff
@@ -19,7 +19,7 @@
       <b>qty:</b><input type="text" class="qty" ng-model="qty"><br>
```

Comment:
Again, you should use the number input type.

Also, above, when you want to display the price, angular has a service called 'filters' where you can give it a value and it will automatically prettify it.

try:   {{ product.price | currency }}

that will use the 'currency' filter